aQute.libg/src/aQute/libg/glob/Glob.java

```diff
@@ -40,65 +51,93 @@ private static String convertGlobToRegEx(String line) {
 		line = line.trim();
 		int strLen = line.length();
 		StringBuilder sb = new StringBuilder(strLen << 2);
-		int inCurlies = 0;
+		int curlyLevel = 0;
+		State state = State.SIMPLE;
+
 		char previousChar = 0;
 		for (int i = 0; i < strLen; i++) {
 			char currentChar = line.charAt(i);
 			switch (currentChar) {
 				case '*' :
-					if (!isEnd(previousChar)) {
+					if ((state == State.SIMPLE || state == State.CURLIES) && !isEnd(previousChar)) {
 						sb.append('.');
 					}
 					sb.append(currentChar);
 					break;
 				case '?' :
-					if (!isStart(previousChar) && !isEnd(previousChar)) {
+					if ((state == State.SIMPLE || state == State.CURLIES) && !isStart(previousChar)
+						&& !isEnd(previousChar)) {
 						sb.append('.');
 					} else {
 						sb.append(currentChar);
 					}
 					break;
 				case '+' :
-					if (!isEnd(previousChar)) {
+					if ((state == State.SIMPLE || state == State.CURLIES) && !isEnd(previousChar)) {
 						sb.append('\\');
 					}
 					sb.append(currentChar);
 					break;
 				case '\\' :
 					sb.append(currentChar);
 					if (i + 1 < strLen) {
-						sb.append(line.charAt(++i));
+						char nextChar = line.charAt(++i);
+						if (state == State.SIMPLE && nextChar == 'Q') {
+							state = State.QUOTED;
+						} else if (state == State.QUOTED && nextChar == 'E') {
+							state = State.SIMPLE;
+						}
+						sb.append(nextChar);
 					}
 					break;
+				case '[' :
+					if (state == State.SIMPLE) {
+						state = State.BRACKETS;
+					}
+					sb.append(currentChar);
+					break;
+				case ']' :
+					if (state == State.BRACKETS) {
+						state = State.SIMPLE;
+					}
+					sb.append(currentChar);
+					break;
+
 				case '{' :
-					if (!isEnd(previousChar)) {
+					if ((state == State.SIMPLE || state == State.CURLIES) && !isEnd(previousChar)) {
+						state = State.CURLIES;
 						sb.append("(?:");
-						inCurlies++;
+						curlyLevel++;
 					} else {
 						sb.append(currentChar);
 					}
 					break;
 				case '}' :
-					if (inCurlies > 0) {
+					if (state == State.CURLIES && curlyLevel > 0) {
 						sb.append(')');
-						inCurlies--;
+						currentChar = ')';
```

Comment:
Perhaps it would be better to add `}` to `isEnd` than changing `currentChar` to get the same effect? I would prefer that.

Reply:
Nope, then you have to know if the } is for  a quantifier. In this case, it is only for a group.